File: config/local_llm.py
"""
로컬 LLM 구성 모듈

LM Studio 등 로컬에서 실행되는 LLM 관련 설정을 정의합니다.
"""

from typing import Dict, List, Optional, Union
from pydantic import BaseModel, Field

DEFAULT_LM_STUDIO_URL = "http://127.0.0.1:1234"

class LocalLLMConfig(BaseModel):
    """로컬 LLM 구성 모델"""
    
    enabled: bool = True
    base_url: str = DEFAULT_LM_STUDIO_URL
    name: str = "LM Studio DeepSeek"
    model_id: str = "deepseek-r1-distill-qwen-7b"
    max_tokens: int = 1000
    temperature: float = 0.7
    top_p: float = 0.9
    timeout: int = 60  # 초


# 기본 로컬 LLM 구성
default_local_llm_config = LocalLLMConfig()

File: frontend/types/local-llm.ts
// 로컬 LLM 관련 타입 정의

// 로컬 LLM 상태 응답
export interface LocalLLMStatus {
  enabled: boolean;
  connected: boolean;
  base_url: string;
  model_id?: string;
  error?: string;
}

// 로컬 LLM 설정
export interface LocalLLMConfig {
  enabled: boolean;
  base_url: string;
  name: string;
  model_id: string;
  max_tokens: number;
  temperature: number;
  top_p: number;
  timeout: number;
}

// 로컬 LLM 설정 업데이트 요청
export interface LocalLLMConfigUpdate {
  enabled?: boolean;
  base_url?: string;
  model_id?: string;
  max_tokens?: number;
  temperature?: number;
  top_p?: number;
}

// 로컬 LLM 채팅 메시지
export interface LocalLLMMessage {
  role: 'user' | 'assistant' | 'system' | 'tool';
  content: string;
  name?: string;
  tool_calls?: any[];
  tool_call_id?: string;
  base64_image?: string;
}

// 로컬 LLM 채팅 요청
export interface LocalLLMChatRequest {
  messages: LocalLLMMessage[];
  system_message?: string;
  max_tokens?: number;
  temperature?: number;
  top_p?: number;
}

// 로컬 LLM 채팅 응답
export interface LocalLLMChatResponse {
  content: string;
  model_id: string;
}